src/relation/mod.rs

```diff
@@ -260,13 +260,17 @@ impl Map {
         }
     }
 
-    /// Compute the schema and exprs of the reduce
-    fn schema_exprs(named_exprs: Vec<(String, Expr)>, input: &Relation) -> (Schema, Vec<Expr>) {
+    /// Compute the schema and exprs of the map
+    fn schema_exprs(named_exprs: Vec<(String, Expr)>, input: &Relation, filter: &Option<Expr>) -> (Schema, Vec<Expr>) {
```

Comment:
For consistency / readability I would keep the args in the same order as in the Map Struct.
`named_exprs`, `filter`, `input`.